sdk/js/__tests__/Client.spec.js

```diff
@@ -56,6 +56,40 @@ describe('DevCycleClient tests', () => {
         expect(client.config).toStrictEqual(testConfig)
     })
 
+    it('should use bootstrapped config when set, not fetch a new one, initialize SDK', async () => {
+        const client = new DevCycleClient(
+            'test_sdk_key',
+            { user_id: 'user1' },
+            { bootstrapConfig: testConfig },
+        )
+        await client.onInitialized
+        expect(getConfigJson_mock).not.toBeCalled()
+        expect(client.config).toStrictEqual(testConfig)
+        expect(client.user.user_id).toEqual('user1')
+    })
+
+    it('should use bootstrapped config when set, ignore cached config', async () => {
+        getConfigJson_mock.mockImplementation(() => {
+            return Promise.resolve({ ...testConfig, cached: true })
+        })
+        const spy = jest.spyOn(window.localStorage.__proto__, 'getItem')
+        const client = new DevCycleClient('test_sdk_key', { user_id: 'user1' })
+        await client.onInitialized
+        // one call to get anon ID, one to get cached config
+        expect(spy).toHaveBeenCalledTimes(2)
+        // construct another client to test if it reads from the cache populated by the initialization of the first
+        // client
+        const client2 = new DevCycleClient(
+            'test_sdk_key',
+            { user_id: 'user1' },
+            { bootstrapConfig: testConfig },
+        )
+        await client2.onInitialized
+        // one more call to get anon ID
+        expect(spy).toHaveBeenCalledTimes(3)
+        expect(client2.config).toStrictEqual(testConfig)
```

Comment:
if the getConfigJson_mock is using a little changed config, we can test to make sure both clients use different configs

Reply:
that's basically what I'm testing here, that client2 has the config provided for boostrapping rather than the modified one that client1 got

Reply:
i missed the `cached: true` 😅